keps/sig-api-machinery/2839-in-use-protection/README.md

```diff
@@ -303,10 +311,38 @@ Therefore, liens field isn't defined as a subresource and providing a separate R
 
 #### Unresolved issues
 
-- [ ] Choice of an advisory feature
-- [ ] Subresource (Is the conclusion that it is needed but will be added later OK?)
-- [ ] Compatibility to existing deletion flows for existing tools (Is current behavior OK and current description enough?)
-- [ ] Name of the field (Is `liens` OK?)
+- Decision: When you delete a namespace and an object in that namespace has a lien:
+  - Does the namespace and object get deleted anyway?
+    - [x] Yes.
+    - [ ] No. See below cases:
+      - Does the namespace deletion get blocked (i.e., the delete request fails) if any object in the namespace has a lien?
+        - No.
+      - Does the other objects in the namespace get deleted, but that not object, preventing the namespace deletion from completing?
+        - [x] No. Namespace lifecycle cleanup will not honor liens. if you delete the namespace, the content will be removed.
+	    - [ ] Yes. See below case:
+          - Does the namespace deletion not proceed until all objects in the namespace are free of liens?
+            - [x] No. Namespace lifecycle cleanup will not honor liens. If you delete the namespace, the content will be removed.
+	        - [ ] ~~Yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't be deleted until all the content will be deleted.~~
+	        - [ ] Partially yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't start deleting the content with liens until all the content without liens will be deleted (Current KEP's description (Also see [Namespace Deletion](#namespace-deletion))).
```

Comment:
I agree that cluster admins should have a power to eventually delete the namespace. However, I would like to make this feature to be able to keep the order of deletion as much as possible. In most use cases, order of deletion is required within a namspace. So, if it isn't kept when deleted via namespace, this feature may become useless. WDYT?

(Please also see [here](https://github.com/kubernetes/enhancements/pull/2840#discussion_r714364698) for use case and [here](https://github.com/kubernetes/enhancements/pull/2840#discussion_r715783357) for an idea to solve it).

Reply:
I think this is important, but it's also a niche case and we could go into Alpha an refine it before beta.

Something like: when deleting a namespace, any resource which fails to delete because of a lien is moved to the back of the list of things-to-delete. If we find that the list only includes things that fail, we leave the NS deletion pending.